ddtrace/contrib/internal/avro/patch.py

```diff
@@ -0,0 +1,74 @@
+import avro
+import wrapt
+
+from ddtrace import config
+from ddtrace.internal.utils.wrappers import unwrap
+from ddtrace.pin import Pin
+
+from .schema_iterator import SchemaExtractor
+
+
+config._add(
+    "avro",
+    dict(),
+)
+
+
+def get_version():
+    # type: () -> str
+    return getattr(avro, "__version__", "")
+
+
+def patch():
+    """Patch the instrumented methods"""
+    if getattr(avro, "_datadog_patch", False):
+        return
+    avro._datadog_patch = True
+
+    _w = wrapt.wrap_function_wrapper
+
+    _w("avro.io", "DatumReader.read", _traced_deserialize)
+    _w("avro.io", "DatumWriter.write", _traced_serialize)
+    Pin(service=None).onto(avro.io.DatumReader)
+    Pin(service=None).onto(avro.io.DatumWriter)
+
+
+def unpatch():
+    if getattr(avro, "_datadog_patch", False):
+        avro._datadog_patch = False
+
+        unwrap(avro.io.DatumReader, "read")
+        unwrap(avro.io.DatumWriter, "write")
+
+
+#
+# tracing functions
+#
+def _traced_serialize(func, instance, args, kwargs):
+    pin = Pin.get_from(instance)
+    if not pin or not pin.enabled():
+        return func(*args, **kwargs)
+
+    active = pin.tracer.current_span()
+
+    try:
+        func(*args, **kwargs)
+    finally:
+        if config._data_streams_enabled and active:
+            SchemaExtractor.attach_schema_on_span(instance.writers_schema, active, SchemaExtractor.SERIALIZATION)
```

Comment:
Makes sense to me, I see you're not creating a new span, same thing as we did in Java. That makes sense to me.